digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala

```diff
@@ -70,17 +71,392 @@ class NodesAndJunctionsService() {
                 }.toMap
               )
             )
-        }.toMap
+        } ++ Seq((None, getTemplatesByBoundingBox(boundingRectangle)))
+        nodesAndJunctions.toMap
+      }
+    }
+  }
+
+  def getTemplatesByBoundingBox(boundingRectangle: BoundingRectangle): (Seq[NodePoint], Map[Junction, Seq[JunctionPoint]]) = {
+    withDynSession {
+      time(logger, "Fetch NodePoint and Junction + JunctionPoint templates") {
+        val junctionPoints = junctionPointDAO.fetchTemplatesByBoundingBox(boundingRectangle)
+        val junctions = junctionDAO.fetchByIds(junctionPoints.map(_.junctionId))
+        val nodePoints = nodePointDAO.fetchTemplatesByBoundingBox(boundingRectangle)
+        (nodePoints, junctions.map { junction => (junction, junctionPoints.filter(_.junctionId == junction.id)) }.toMap)
+      }
+    }
+  }
+
+  def handleJunctionPointTemplates(projectLinks: Seq[ProjectLink]): Unit = {
```

Comment:
We should not have methods with 200 lines of code.
We need to split this method and try to avoid repeated code

Reply:
true, should be treated in proper PR (US 1843-> task 1938)

Reply:
US 1843 code is already included in this PR. This needs to be fixed here